main.cpp

```diff
@@ -180,7 +191,14 @@ std::vector<Element> Storage::GetLogs(time_t start, time_t finish)
         {
             if (elem.GetTime() >= start && elem.GetTime() <= finish)
             {
-                dates.push_back(elem);
+                if(dates.empty()||dates[0].GetTime()<elem.GetTime())
+                {
+                    dates.push_back(elem);
+                }
+                else
+                {
+                    dates.insert(dates.begin(), elem);
```

Comment:
Similar here, only this line will execute, not #196. Mixing the order of Elements - those from  _new will be arranged descending, while ones from _old will be ascending.

Reply:
I was looking in the wrong direction 0_0. Now it's fixed , and GetLogs() works with GetPos() instead of any logic. And returning vector  has ascending order, too.